src/snowflake/snowpark/modin/plugin/extensions/series_overrides.py

```diff
@@ -376,44 +372,108 @@ def __init__(
     # use this list to update inplace when there is a shallow copy.
     self._siblings = []
 
-    # modified:
-    # Engine.subscribe(_update_engine)
+    from snowflake.snowpark.modin.plugin.extensions.index import Index
+
+    if query_compiler:
+        # CASE 1: query_compiler
+        # If a query_compiler is passed in, only use the query_compiler and name fields to create a new Series.
+        self._query_compiler = query_compiler.columnarize()
+        if name is not None:
+            self.name = name
+        return
```

Comment:
I don't think it's a good idea to move most of the series logic into the dataframe constructor. They share some logic but that is mostly the set_index/reindex logic. I moved that out into helper methods. I still think it's a good idea to keep the "general" cases for series and df separately since the series logic is a lot simpler than df's.